internal/api/auth/basic_auth.go

```diff
@@ -12,6 +14,43 @@ const (
 	realm                = `Basic realm="restricted"`
 )
 
+type authMode int
+
+const (
+	NoAuth authMode = iota
+	LDAPMode
+)
+
+type BasicAuth struct {
+	ldapAuth *LDAPAuth
+	mode     authMode
+}
+
+func NewBasicAuth(mode authMode) BasicAuth { return BasicAuth{mode: mode} }
+
+func (b *BasicAuth) ConfigureLdap(ldap *LDAPAuth) error {
+	if ldap == nil {
+		return errors.New("LDAP configuration is missing")
+	}
+	b.ldapAuth = ldap
+
+	return nil
+}
+
+func (b *BasicAuth) Wrap(next httprouter.Handle) httprouter.Handle {
+	switch b.mode {
+	case NoAuth:
+		return func(w http.ResponseWriter, r *http.Request, ps httprouter.Params) { next(w, r, ps) }
+	case LDAPMode:
+		return BasicAuthLDAP(b.ldapAuth, next)
+	default:
+		return func(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+			log.Error().Str("authentication issue", "bad server configuration").Send()
```

Comment:
We might want to also log the value of `b.mode` here

Reply:
done